internal/scan-manager/scanners/veracode/src/main/java/org/wso2/security/tools/scanmanager/scanners/veracode/service/VeracodeScanner.java

```diff
@@ -0,0 +1,304 @@
+        VeracodeScannerConfiguration.getInstance().setLogFilePath(YAMLConfigurationReader.getInstance().
+                getConfigProperty(VeracodeScannerConstants.VERACODE_LOG_FILE_PATH));
+        VeracodeScannerConfiguration.getInstance().setScannerClass(YAMLConfigurationReader.getInstance().
+                getConfigProperty(VeracodeScannerConstants.SCANNER_BEAN_CLASS_NAME));
+    }
+
+    /**
+     * Run the scan using product zip file.
+     *
+     * @param scanRequest Object that represent the required information for the scanner operation
+     * @return ResponseEntity with status of the request
+     */
+    @Override
+    public ResponseEntity startScan(ScannerScanRequest scanRequest) {
+        scanContext = new ScanContext();
+        scanContext.setJobId(scanRequest.getJobId());
+        scanContext.setAppId(scanRequest.getAppId());
+        scanContext.setArtifactLocation(scanRequest.getFileMap().get(VeracodeScannerConstants.SCAN_ARTIFACT).get(0));
+        ResponseEntity responseEntity;
+
+        if (scanRequest.getFileMap().get(VeracodeScannerConstants.SCAN_ARTIFACT) != null) {
+            if (!StringUtils.isEmpty(scanContext.getAppId())) {
+                responseEntity = new ResponseEntity<>(HttpStatus.ACCEPTED);
+                startVeracodeScan();
+            } else {
+                String message = "Error occured while submitting the start scan request since " +
+                        "the application is empty in the request. ";
+                responseEntity = new ResponseEntity<>(new ErrorMessage(HttpStatus.BAD_REQUEST.value(),
+                        message), HttpStatus.BAD_REQUEST);
+                CallbackUtil.updateScanStatus(scanContext.getJobId(), ScanStatus.ERROR, null,
+                        null);
+                CallbackUtil.persistScanLog(scanContext.getJobId(), message, LogType.ERROR);
+            }
+        } else {
+            String message = "Error occured while submitting the start scan request since " +
+                    "the scan artifacts are empty in the request. ";
+            responseEntity = new ResponseEntity<>(new ErrorMessage(HttpStatus.BAD_REQUEST.value(),
+                    message), HttpStatus.BAD_REQUEST);
+            CallbackUtil.updateScanStatus(scanContext.getJobId(), ScanStatus.ERROR, null,
+                    null);
+            CallbackUtil.persistScanLog(scanContext.getJobId(), message, LogType.ERROR);
+        }
+        return responseEntity;
+    }
+
+    /**
+     * Stop the last scan for a given application.
+     *
+     * @param scanRequest Object that represent the required information for tha scanner operation
+     * @return ResponseEntity with status of the request
+     */
+    @Override
+    public ResponseEntity cancelScan(ScannerScanRequest scanRequest) {
+
+        ResponseEntity responseEntity;
+        String result;
+
+        try {
+            result = uploadAPIWrapper.deleteBuild(scanContext.getAppId());
```

Comment:
Shouldn't we check if we have already started the scan in `actual scanner` or not before doing this?